src/Icicle/Repl.hs

```diff
@@ -6,80 +6,94 @@ module Icicle.Repl (
   , sourceCheck
   , sourceConvert
   , sourceParseConvert
+  , readFacts
   ) where
 
-import qualified        Icicle.Common.Fresh             as Fresh
-import qualified        Icicle.Common.Base              as Com
-import qualified        Icicle.Common.Type              as Com
-import qualified        Icicle.Core.Program.Program     as Core
-
-import qualified        Icicle.Data                     as D
-import qualified        Icicle.Dictionary               as D
-
-import qualified        Icicle.Source.Checker.Checker   as SC
-import qualified        Icicle.Source.Checker.Error     as SC
-import qualified        Icicle.Source.Parser            as SP
-import qualified        Icicle.Source.Query             as SQ
-import qualified        Icicle.Source.ToCore.ToCore     as STC
-import qualified        Icicle.Source.ToCore.Base       as STC
-import qualified        Icicle.Source.Type              as ST
-
-import                  Icicle.Internal.Pretty
-
-import                  P
-
-import                  Data.Either.Combinators
-
-import qualified        Data.Map                        as Map
-import qualified        Data.Text                       as T
+import           Icicle.BubbleGum
+import           Icicle.Common.Base
+import qualified Icicle.Common.Fresh           as Fresh
+import           Icicle.Common.Type
+import qualified Icicle.Core.Program.Program   as Core
+import           Icicle.Data
+import qualified Icicle.Dictionary             as D
+import           Icicle.Internal.Pretty
+import qualified Icicle.Serial                 as S
+import qualified Icicle.Source.Checker.Checker as SC
+import qualified Icicle.Source.Checker.Error   as SC
+import qualified Icicle.Source.Parser          as SP
+import qualified Icicle.Source.Query           as SQ
+import qualified Icicle.Source.ToCore.Base     as STC
+import qualified Icicle.Source.ToCore.ToCore   as STC
+import qualified Icicle.Source.Type            as ST
+import qualified Icicle.Simulator as S
+import qualified Icicle.Core.Eval.Program as EP
+
+import           P
+
+import           Data.Either.Combinators
+import qualified Data.Map                      as Map
+import           Data.Text                     (Text)
+import qualified Data.Text                     as T
+import qualified Data.Traversable              as TR
```

Comment:
/nitpick usually we try and order imports alphabetically and divide up on prefix.
eg
```
import           Data.Either.Combinators

import           Icicle.BubbleGum
```